adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapter.java

```diff
@@ -96,4 +112,38 @@ protected String getTypeName() {
         }
         return result.future();
     }
+
+    private Future<ResourceIdentifier> mapMessage(final MqttContext ctx, final ResourceIdentifier targetAddress, final MqttPublishMessage message) {
+        return getRegistrationAssertion(targetAddress.getTenantId(), targetAddress.getResourceId(),
+                ctx.authenticatedDevice(), ctx.getTracingContext()).compose(registratonInfo -> {
+            final Promise<ResourceIdentifier> result = Promise.promise();
+            if (!registratonInfo.containsKey(RegistrationConstants.FIELD_MAPPER)) {
+                result.complete(targetAddress);
+                return result.future();
+            }
+            final String mapper = registratonInfo.getString(RegistrationConstants.FIELD_MAPPER);
+            if (mapper.isBlank()) {
+                result.complete(targetAddress);
+                return result.future();
+            }
+            final HashMap<String, MapperEndpoint> mapperEndpoints = getConfig().getMapperEndpoints();
```

Comment:
Do we really need to configure the mapping endpoints *per device*? Will they be different for all devices or can't we simply configure the (limited set of) mapper endpoints at the tenant level instead?

Reply:
In our case, we might have different mapping endpoint per device in the same tenant.
Also, I thought this configuration is on the protocol adapter (which supports multiple tenants) so a list of mapperEndpoints should be injected here or am I wrong?

Reply:
I see. However, instead of exposing the Map to this class here, what about adding a *getMapperEndpoint(String)* to the `ProtocolAdapterProperties` class instead?

Reply:
Done